core/arch/arm/kernel/entry_a32.S

```diff
@@ -447,6 +448,39 @@ shadow_stack_access_ok:
 	bls	shadow_stack_access_ok
 #endif
 
+#if defined(CFG_DYN_STACK_CONFIG)
+	ldr	r0, =boot_embdata_ptr
+	ldr	r0, [r0]
+	sub	r1, r0, #THREAD_BOOT_INIT_TMP_ALLOC
+	sub	r0, r0, #(__STACK_TMP_OFFS + __STACK_CANARY_SIZE)
```

Comment:
Testing on my 32b HW, I saw an issue here: I think it is due to `thread_core_local` structure not being zero initialized here.

My OP-TEE case panics during drivers inits `assert_have_no_spinlock()` the first time a mutex is locked. Digging a bit I saw that it was due to field `thread_core_local::locked_count` that contains a surprising high value, likely because is was not zero initialized.

I think it would be worth to zero initialize the whole `struct thread_core_local` here. At least it fixed my boot issue. I think the same should apply on Aarch64.

Reply:
Yes, that makes sense. I'll add something.